### PR TITLE
feat: serialization for QuantizedTensor

### PR DESCRIPTION
**Pull request by i8run**
## What changes were proposed in this pull request?

QuantizedTensor has a byte array which stores the quantized value.
So we add byte supporting in bigdl.proto.

QuantizedTensor is not like other DenseTensor, so the logic in
DataConverter should be modified for supporting serializing it.

## How was this patch tested?

Jenkins.

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala [hunk 1]
```diff
@@ -352,20 +405,56 @@ object DataConverter extends DataConverter{
...
         } else {
           val storageBuilder = TensorStorage.newBuilder
           if (tensorNumeric == NumericFloat) {
-            tensorBuilder.setDatatype(DataType.FLOAT)
-            storageBuilder.setDatatype(DataType.FLOAT)
-            tensor.storage().array().asInstanceOf[Array[Float]].
-              foreach(data => storageBuilder.addFloatData(data))
+            tensor.getTensorType match {
+              case DenseType =>
+                tensorBuilder.setDatatype(DataType.FLOAT)
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala]: why put setDataType inside case clause ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala [hunk 2]
```diff
@@ -352,20 +405,56 @@ object DataConverter extends DataConverter{
...
+                storageBuilder.setDatatype(DataType.FLOAT)
+                tensor.storage().array().asInstanceOf[Array[Float]].
+                  foreach(data => storageBuilder.addFloatData(data))
+              case QuantizedType =>
+                tensorBuilder.setDatatype(DataType.FLOAT)
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala]: why put setDataType inside case clause ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala [hunk 3]
```diff
@@ -431,17 +520,21 @@ object DataConverter extends DataConverter{
...
         } else {
-          val tensorNumeric = tensor.getTensorNumeric()
-          val offSet = tensor.storageOffset()
           val totalElement = tensor.nElement()
           val dimension = tensor.dim()
-          val isScalar = tensor.isScalar
           val tensorBuilder = BigDLTensor.newBuilder
           tensorBuilder.setId(tensorId)
           tensorBuilder.setDimension(dimension)
-          tensorBuilder.setOffset(offSet)
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala]: why remove this ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala [hunk 4]
```diff
@@ -431,17 +520,21 @@ object DataConverter extends DataConverter{
           attributeBuilder.setTensorValue(storages.get(tensorId).get
             .asInstanceOf[BigDLTensor])
         } else {
-          val tensorNumeric = tensor.getTensorNumeric()
-          val offSet = tensor.storageOffset()
           val totalElement = tensor.nElement()
           val dimension = tensor.dim()
-          val isScalar = tensor.isScalar
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala]: why remove this ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala (hunk 3 above)
Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala]: For `QuantizedTensor`, there's no such methods supporting. So there're in next lines for `DenseTensor`.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala (hunk 4 above)
Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala]: It seems as previous.

**Pull request merged**